Zend/zend_vm_def.h

```diff
@@ -6188,7 +6188,7 @@ ZEND_VM_HANDLER(77, ZEND_FE_RESET_R, CONST|TMP|VAR|CV, JMP_ADDR)
 			}
 		}
 	} else {
-		zend_error(E_WARNING, "Invalid argument supplied for foreach()");
+		zend_error(E_WARNING, "Argument supplied for foreach() must be of type array|object, %s given", zend_get_type_by_const(Z_TYPE_P(array_ptr)));
```

Comment:
How about `foreach() argument must be of type ...`?

Reply:
hmm, somehow it doesn't feel very natural to me. Let's ask around :)

Reply:
What do you think about following the PR #5198, where we have eg.
`array_diff_assoc(): Argument #1 ($arr1) must be of type array, %s given`
and go for the following form?
`foreach(): Argument #1 ($arr) must be of type array, %s given`

Reply:
I think this message is misleading because it looks like so that `foreach` is a function. Plus I don't think showing a parameter name makes sense in this case. So most probably we should choose between the two previous options.

Reply:
Yeah looking at it that way, I agree. Between the previous two options I lean more towards `foreach() argument`, it's simpler :-)

Reply:
Changed!